ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/LedgerApiBenchTool.scala

```diff
@@ -38,38 +42,60 @@ object LedgerApiBenchTool {
-    val printer = pprint.PPrinter(200, 1000)
-    logger.info(s"Starting benchmark with configuration:\n${printer(config).toString()}")
+    logger.info(s"Starting benchmark with configuration:\n${prettyPrint(config)}")
 
     implicit val resourceContext: ResourceContext = ResourceContext(ec)
 
     apiServicesOwner(config).use { apiServices =>
-      def testContractsGenerationStep(): Future[Unit] = config.contractSetDescriptorFile match {
-        case None =>
-          Future.successful(
-            logger.info("No contract set descriptor file provided. Skipping contracts generation.")
+      def benchmarkStep(streams: List[Config.StreamConfig]): Future[Unit] =
+        if (streams.isEmpty) {
+          Future.successful(logger.info(s"No streams defined. Skipping the benchmark step."))
+        } else {
+          Benchmark.run(
+            streams = streams,
+            reportingPeriod = config.reportingPeriod,
+            apiServices = apiServices,
+            metricsReporter = config.metricsReporter,
           )
+        }
+
+      def submissionStep(
+          submissionDescriptor: SubmissionDescriptor
+      ): Future[CommandSubmitter.SubmissionSummary] =
+        CommandSubmitter(apiServices).submit(
+          descriptor = submissionDescriptor,
+          maxInFlightCommands = config.maxInFlightCommands,
+          submissionBatchSize = config.submissionBatchSize,
+        )
+
+      config.contractSetDescriptorFile match {
+        case None =>
+          benchmarkStep(config.streams)
         case Some(descriptorFile) =>
-          CommandSubmitter(apiServices).submit(
-            descriptorFile,
-            config.maxInFlightCommands,
-            config.submissionBatchSize,
-          )
-      }
-
-      def benchmarkStep(): Future[Unit] = if (config.streams.isEmpty) {
-        Future.successful(logger.info(s"No streams defined. Skipping the benchmark step."))
-      } else {
-        Benchmark.run(config, apiServices)
+          for {
+            descriptor <- Future.fromTry(parseDescriptor(descriptorFile))
+            _ = logger.info(prettyPrint(descriptor))
+            summary <- submissionStep(descriptor.submission)
+            streams = descriptor.streams.map(
+              DescriptorConverter.streamDescriptorToConfig(_, summary)
+            )
+            _ = logger.info(s"Converted stream configs: ${prettyPrint(streams)}")
+            _ <- benchmarkStep(streams)
+          } yield ()
       }
-
-      for {
-        _ <- testContractsGenerationStep()
-        _ <- benchmarkStep()
-      } yield ()
     }
   }
 
+  private def parseDescriptor(descriptorFile: File): Try[WorkflowDescriptor] =
+    SimpleFileReader.readFile(descriptorFile)(WorkflowParser.parse).flatMap {
+      case Left(err: WorkflowParser.ParserError) =>
+        Failure(
+          CommandSubmitter.CommandSubmitterError(s"Workflow parsing error. Details: ${err.details}")
```

Comment:
This error seems to be misleading. We need a new set of error case classes for parsing, submissions and benchmarking

Reply:
I've changed this to use `RuntimeException()` because the only thing we do with exception at this point is to print details.